zzz_generated/all-clusters-app/zap-generated/PluginApplicationCallbacks.h

```diff
@@ -1,88 +0,0 @@
-/*
```

Comment:
How do we ensure that the new generated code matches the old (or more generally for future changes to this generator how do we adequately review changes to the generated code)?

Reply:
I have manually validated that the files are perfectly equivalent for a random choice of items (I believe lock, light and all-clusters ... lost the actual list while squash-merging but it was at least 3).

If the items are wrong, it would currently fail to compile due to undefined names. This thing gets expanded into a list of function calls and if the functions are not defined it would fail to compile. So we cover the 'extra/mis-named items' quite well, however we do not cover the 'missing items' as well. We did not cover those during code review as much either though and we tend to just trust zap to generate things. Here we trust codegen.py and we have the initial manual comparison/review.

Reply:
We also have a unit test now that validates on a known input if the known output matches. The known input however is indeed on the smaller side (unit tests are intended to be small to be easily reviewed).

Reply:
> We did not cover those during code review as much either though

I absolutely do, fwiw.  And we have absolutely caught bugs in both templates and core codegen machinery by looking at the diff to generated code, especially when that diff was unexpected.

At the very least, we should have some mechanism for verifying that changes that are expected to not change any generated code (e.g. refactoring of the generator) in fact do not change it....  Hard to do in CI, but making it easy to do that manually would be good.

Reply:
Unit test in `scripts/idl/tests/outputs/several_clusters/cpp-app/` validate that no unexpected changes. It only has client callbacks though for now. 

Are you asking for a more complex example (would one be sufficient to feel more comfortable with these)?

Reply:
For manual delta checks, codegen.py can be invoked manually so one can checkout different versions and check what codegen would output. Somewhat tedious though.

Reply:
Right, I am asking whether we can automate the tedious bit.

This is not just about this specific file in this specific PR; this file is simple enough that I am fairly confident the changes are correct based on code inspection and the checking that has already happened.  This is a question about how we sanity-check PRs going forward.

Reply:
I don't know that there is an easy answer there honestly. Having small-ish representative 'unit-tests' for each code-generation output format and having the resultant generated code be checked into the tree for that small unit-test would be one approach.

That's the approach I had taken when building the Weave schema compiler - I check in a reference generated code along with the templates alongside each other (see [this](https://github.com/openweave/openweave-wdlc/tree/master/codegen/weave-device-cpp)).

Reply:
I updated the code to generate both client and server clusters in the `several_clusters` unit test, this should validate all codegen options for the plugin code, EXCEPT if the list is empty, which case is not handled but probably not relevant unless we start removing things like client callbacks.